src/hip_module.cpp

```diff
@@ -1454,21 +1455,7 @@ hipError_t hipLaunchKernel(
    hipFunction_t kd = hip_impl::get_program_state().kernel_descriptor((std::uintptr_t)func_addr,
                                                            hip_impl::target_agent(stream));
 
-   if(kd == nullptr || kd->_header == nullptr)
-       return ihipLogStatus(hipErrorInvalidValue);
-
-   size_t szKernArg = kd->_header->kernarg_segment_byte_size;
-
-   if(args == NULL && szKernArg != 0)
-      return ihipLogStatus(hipErrorInvalidValue);
-
-   void* config[]{
-        HIP_LAUNCH_PARAM_BUFFER_POINTER,
-        args,
-        HIP_LAUNCH_PARAM_BUFFER_SIZE,
-	    &szKernArg,
-        HIP_LAUNCH_PARAM_END};
-
-   return ihipLogStatus(ihipModuleLaunchKernel(tls, kd, numBlocks.x * dimBlocks.x, numBlocks.y * dimBlocks.y, numBlocks.z * dimBlocks.z,
-                          dimBlocks.x, dimBlocks.y, dimBlocks.z, sharedMemBytes, stream, nullptr, (void**)&config, nullptr, nullptr, 0));
+   return hipModuleLaunchKernel(kd, numBlocks.x, numBlocks.y, numBlocks.z,
```

Comment:
@scchan, I think we don't want to call two external HIP APIs one in side other, like in this PR hipLaunchKernel calling hipModuleLaunchKernel. May I know the reason for calling it?

Reply:
The original implementation has an incorrect assumption that all every array element from args is a pointer to a gpu buffer.  Also, the kernel arg should be constructed based on the metadata of the kernel.  I see hipLaunchKernel as a simple wrapper around hipModuleLaunchKernel.